scripts/builtin/intersection.dml

```diff
@@ -0,0 +1,57 @@
+#-------------------------------------------------------------
+#
+# Copyright 2020 Graz University of Technology
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+# Implements set intersection for numeric data  
+
+
+# INPUT PARAMETERS:
+# ---------------------------------------------------------------------------------------------
+# NAME            TYPE    DEFAULT     MEANING
+# ---------------------------------------------------------------------------------------------
+# X               Double  ---         matrix X, set A
+# Y               Double  ---         matrix Y, set B
+# ---------------------------------------------------------------------------------------------
+ 
+#Output(s)
+# ---------------------------------------------------------------------------------------------
+# NAME                      TYPE    DEFAULT      MEANING
+# ---------------------------------------------------------------------------------------------
+# intersectionSet           Double    ---         intersection matrix, set of intersecting items
+# intersectCount            Integer               Number of intersecting  items
+
+m_intersection = function(Matrix[Double] X, Matrix[Double] Y)
+  return(Matrix[Double] intersectionSet, Integer intersectCount)
+{
+  X = (table(X, 1) != 0 ) 
+  X = X * seq(1, nrow(X))
+  Y = (table(Y, 1) != 0 ) 
+  Y = Y * seq(1, nrow(Y))
+  
+  n = ifelse(nrow(X) < nrow(Y), nrow(X), nrow(Y))
+  R = (X[1: n,] * Y[1:n,]) !=0
+  R = R* seq(1,n)
+  R = removeEmpty(target=R, margin="rows")
```

Comment:
by feeding the original R into the optional select argument of removeEmpty we can avoid an unnecessary pass.